servers/plugins/org.fusesource.ide.server.karaf.core/src/org/fusesource/ide/server/karaf/core/runtime/KarafRuntimeLocator.java

```diff
@@ -144,7 +144,8 @@ public IRuntimeWorkingCopy getRuntimeFromDir(File dir, IProgressMonitor monitor)
 			ServerBeanType type = sb.getBeanType();
 			if( type != null ) {
 				if( type.equals(KarafBeanProvider.KARAF_2x) || 
-					type.equals(KarafBeanProvider.KARAF_3x) ) {
+					type.equals(KarafBeanProvider.KARAF_3x) ||
+					type.equals(KarafBeanProvider.KARAF_4x)) {
```

Comment:
better to invert equals and it might avoid the need to check for nullity of type just before